src/net_processing.h

```diff
@@ -17,6 +17,10 @@ static const int64_t ORPHAN_TX_EXPIRE_TIME = 20 * 60;
 static const int64_t ORPHAN_TX_EXPIRE_INTERVAL = 5 * 60;
 /** Default number of orphan+recently-replaced txn to keep around for block reconstruction */
 static const unsigned int DEFAULT_BLOCK_RECONSTRUCTION_EXTRA_TXN = 100;
+/** Headers download timeout expressed in microseconds
+ *  Timeout = base + per_header * (expected number of headers) */
+static constexpr int64_t HEADERS_DOWNLOAD_TIMEOUT_BASE = 15 * 60 * 1000000; // 15 minutes
```

Comment:
I just wanted to do something clearly safe (it's easy to reason about booting your sync peer potentially every 15 minutes, but slightly scarier to think there could be a circumstance you'd boot that peer every 1 minute).

Anyway I'd like to see what other opinions are, I'm fine adjusting them however people would like...